examples/gallery/main.cpp

```diff
@@ -5,5 +5,65 @@
 
 int main()
 {
-    App::create()->run();
+    auto demo = App::create();
+
+    auto row_data = std::make_shared<slint::VectorModel<
+            std::shared_ptr<slint::Model<slint::private_api::StandardListViewItem>>>>();
```

Comment:
One detail I noticed is that this is using `slint::private_api`. In Rust it seems that `StandardListViewItem` is a public type, but it doesn't look like it is in C++.

Reply:
Yes that's true, I think also it should be public for C++. Shall we change it for now or do it as separate task?

Reply:
In my opinion this can be a separate change, not in this PR. Olivier, what do you think?

Reply:
Yeah, we shouldn't need to use private API in our examples.
It doesn't need to be in this PR but should be done quickly afterwards

Reply:
Ok I will do it after this PR is finished. Any more feedback?